### PR TITLE
Add support for specific test cases

### PR DESCRIPTION
**Pull request by mrchapp**

### CODE REVIEW

File: submit_for_testing.py [hunk 1]
```diff
@@ -290,12 +307,25 @@ def main():
...
+    test_list = []
+    if args.test_plan:
+        test_plan_path = os.path.join(args.testplan_path, args.test_plan)
+        for test in _get_test_plan_list(test_plan_path):
+            test_list.append(os.path.join(test_plan_path, test))
+
+    if args.test_case:
+        for test in args.test_case:
+            test_case_path = os.path.join(args.testcase_path, test)
+            test_list.append(test_case_path)
```
Comment (mwasilew) [inline submit_for_testing.py]: is this intentional to use both --test-plan and --test-case together? Code allows for that.

Comment (mrchapp) [reply submit_for_testing.py]: Added to exclusive group.

**Comment by mwasilew** [comment]
I checked the following command line option:
./submit_for_testing.py --variables variables.ini --device-type x15 --dry-run --test-case bt.yaml --test-lava-validity

and it produces:
```
[            main() ] tmp/x15/bt.yaml
Traceback (most recent call last):
  File "./submit_for_testing.py", line 446, in <module>
    main()
  File "./submit_for_testing.py", line 369, in main
    for test in _get_test_plan_list(test_plan_path):
UnboundLocalError: local variable 'test_plan_path' referenced before assignment
```
The problem happens in line 313 where test_plan_path is only created when --test-plan is used.

**Comment by mwasilew** [comment]
I fixed this issue and added a few more tests. Merged manually so closing this PR.

**Pull request closed without merging**